cli/command.go

```diff
@@ -0,0 +1,295 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/cheynewallace/tabby"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+// RunCMD provides a simple user-centered command-line interface for
+// running commands on a remote instance.
+func RunCMD() cli.Command {
+	const (
+		commandFlagName = "command"
+		envFlagName     = "env"
+		sudoFlagName    = "sudo"
+		sudoAsFlagName  = "sudoAs"
+		idFlagName      = "id"
+		noShellFlagName = "noShell"
+		tagFlagName     = "tag"
+		waitFlagName    = "wait"
+	)
+
+	return cli.Command{
+		Name:  "run",
+		Usage: "Run a command with Jasper",
+		Flags: append(clientFlags(),
+			cli.StringSliceFlag{
+				Name:  joinFlagNames(commandFlagName, "c"),
+				Usage: "specify a command to run on a remote jasper service. may specify more than once",
+			},
+			cli.StringSliceFlag{
+				Name:  envFlagName,
+				Usage: "specify environment variables, in '<key>=<val>' forms. may specify more than once",
+			},
+			cli.BoolFlag{
+				Name:  sudoFlagName,
+				Usage: "use this flag to run the command with sudo",
+			},
+			cli.StringFlag{
+				Name:  sudoAsFlagName,
+				Usage: "use this to run commands as another user as in 'sudo -u <user>'",
+			},
+			cli.StringFlag{
+				Name:  idFlagName,
+				Usage: "specify an id for this process (optional)",
+			},
+			cli.BoolTFlag{
+				Name:  noShellFlagName,
+				Usage: "when specified execute commands directly without shell. If multiple commands are specified they do not share a shell environment",
+			},
+			cli.StringSliceFlag{
+				Name:  tagFlagName,
+				Usage: "specify one or more tag names for the process",
+			},
+			cli.BoolFlag{
+				Name:  waitFlagName,
+				Usage: "specify to block until the process returns (subject to service timeouts), propogating the exit code from process",
+			},
+		),
+		Before: mergeBeforeFuncs(clientBefore(),
+			func(c *cli.Context) error {
+				if len(c.StringSlice(commandFlagName)) == 0 {
+					if c.NArg() == 0 {
+						return errors.New("must specify a command")
+					}
+					return errors.Wrap(c.Set(commandFlagName, strings.Join(c.Args().Tail(), " ")), "problem setting command")
+				}
+				return nil
+			}),
+		Action: func(c *cli.Context) error {
+			envvars := c.StringSlice(envFlagName)
+			cmds := c.StringSlice(commandFlagName)
+			useSudo := c.Bool(sudoFlagName)
+			sudoAs := c.String(sudoAsFlagName)
+			useShell := c.BoolT(noShellFlagName)
```

Comment:
This should be either `useShell := !c.BoolT(noShellFlagName)` or `noShell := c.BoolT(noShellFlagName)`. Otherwise, the only case in which `useShell` is false will be if you set `--noShell=false`.